hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/state/HdfsZkStore.java

```diff
@@ -0,0 +1,102 @@
+package org.apache.mesos.hdfs.state;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.mesos.hdfs.config.HdfsFrameworkConfig;
+import org.apache.mesos.state.State;
+import org.apache.mesos.state.Variable;
+import org.apache.mesos.state.ZooKeeperState;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Manages persistence state of the application.  It does NOT have an references to Protocol Buffers.  It
+ * throws exceptions to allow the manager to handle the logic.  It currently is tied to zookeeper, but should
+ * be replaceable with any mesos state abstraction.
+ */
+public class HdfsZkStore implements HdfsStore {
+
+  private State state;
+  private static final String FRAMEWORK_ID_KEY = "frameworkId";
+
+
```

Comment:
yes.. especially if there is an expectation of replacing the zk store with another type of store... good point.